arrows/ocv/tests/test_inpaint.cxx

```diff
@@ -0,0 +1,108 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include <test_gtest.h>
+
+#include <arrows/ocv/image_container.h>
+#include <arrows/ocv/image_io.h>
+#include <arrows/ocv/inpaint.h>
+
+#include <vital/plugin_loader/plugin_manager.h>
+
+#include <opencv2/core/core.hpp>
+
+#include <gtest/gtest.h>
+
+namespace kv = kwiver::vital;
+namespace ka = kwiver::arrows;
+
+kv::path_t g_data_dir;
+static std::string test_image_name = "images/small_logo_color.png";
+static std::string test_mask_name = "images/small_logo_mask.png";
+
+static std::string expected_telea_result_file = "images/inpaint_telea.png";
+static std::string expected_navier_stokes_result_file =
+  "images/inpaint_navier_stokes.png";
+static std::string expected_mask_result_file = "images/inpaint_mask.png";
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  TEST_LOAD_PLUGINS();
+
+  GET_ARG( 1, g_data_dir );
+
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+class inpaint : public ::testing::Test
+{
+  void SetUp();
+
+  TEST_ARG( data_dir );
+
+  ka::ocv::image_io io;
+  ka::ocv::inpaint inpainter;
+  kv::image_container_sptr input_image;
+  kv::image_container_sptr input_mask;
+  kv::config_block_sptr config;
+
+public:
+  void test_inpaint_type( kv::config_block_sptr const& config,
+                          kv::path_t expected_basename );
+};
+
+// ----------------------------------------------------------------------------
+void
+inpaint
+::SetUp()
+{
+  auto const image_file = data_dir + "/" + test_image_name;
+  auto const mask_file = data_dir + "/" + test_mask_name;
+
+  input_image = io.load( image_file );
+  input_mask = io.load( mask_file );
+
+  config = kv::config_block::empty_config();
+}
+
+// ----------------------------------------------------------------------------
+void
+inpaint
+::test_inpaint_type( kv::config_block_sptr const& config,
+                     kv::path_t expected_basename )
+{
+  inpainter.set_configuration( config );
+
+  auto const inpainted_image_ptr = inpainter.merge( input_image, input_mask );
+
+  auto expected_filename = data_dir + "/" + expected_basename;
+  auto const expected_image_ptr = io.load( expected_filename );
```

Comment:
Would it be worth asserting that the images loaded successfully?

Reply:
It seems like the traceback would be fairly self explanatory if image loading failed. I'm open to adding this check but it seems like a better solution might be to add an `ocv::image_io` unit test as done in `vxl`.

Reply:
I'm not especially worried about the traceback, as we aren't testing image I/O. I'm more worried about the test blowing up in unnecessarily spectacular fashion (e.g. SEGV, which it looks like *will* happen) if, say, one of the images passed to `inpainter.merge` is a null pointer. But I'm also not sure it's even worth worrying that the testing environment has gone sideways, which is why the original comment is a question rather than a suggestion. (Note: if I was going to do something, I would probably `ASSERT_NE(nullptr, input_image)`, and so forth.)

To be clear, I don't particularly object to `inpainter.merge` crashing if you give it null pointers. I rather doubt we tend to check that sort of thing. Don't do that. Usually, we expect the *user* to ensure they don't do that. In this case, the user is the unit test, so one might argue for a paranoia check that we actually did load the images. But, again, something is *really* wrong if that happen, possibly to the extent that "it broke; here's the pieces" is a satisfactory reaction.

Reply:
I don't have strong feelings any way here. However, if we make a decisions to validate image loading there's 10ish `vxl` tests that need to be updated. Which makes be wonder if this is better served as a separate discussion.